isascii not declared in this scope

< Please summarize your issue here.  Check [x] as many boxes below as you can before submitting your issue.  Thanks -- you rock! >

On macOSX El Capitan with gcc-arm-none-eabi-5_3-2016q1 I am getting the error 
`../wiring/inc/spark_wiring_character.h:49:31: error: 'isascii' was not declared in this scope`
`/wiring/inc/spark_wiring_character.h:104:21: error: 'toascii' was not declared in this scope`
running make clean all
---

Completeness:
- [ x] Minimum test case added
- [ x] Device, system and user firmware versions stated
- [x] Particle confirmed

--- a/build/module.mk
+++ b/build/module.mk
@@ -33,6 +33,10 @@ CFLAGS += $(addprefix -D,$(GLOBAL_DEFINES))
 export GLOBAL_DEFINES
 endif
 
+# fixes build errors on ubuntu with arm gcc 5.3.1
+# GNU_SOURCE is needed for isascii/toascii
+# WINSOCK_H stops select.h from being used which conflicts with CC3000 headers
+CFLAGS += -D_GNU_SOURCE -D_WINSOCK_H
 
 # Collect all object and dep files
 ALLOBJ += $(addprefix $(BUILD_PATH)/, $(CSRC:.c=.o))

--- a/platform/NET/CC3000/CC3000_Host_Driver/socket.h
+++ b/platform/NET/CC3000/CC3000_Host_Driver/socket.h
@@ -35,6 +35,13 @@
 #ifndef __SOCKET_H__
 #define __SOCKET_H__
 
+// exclude timeval definition on linux
+#define _SYS__TIMEVAL_H_
+// exclude system select
+#define _SYS_SELECT_H
+
+
+
 #include "data_types.h"
 #include "time.h"
 

